pkg/lenra/manifest.go

```diff
@@ -0,0 +1,147 @@
+package lenra
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Manifest interface {
+	Widgets() []Widget
+	Actions() []Action
+	Resources() []Resource
+}
+type BaseManifest struct{}
+
+var _ Manifest = &BaseManifest{}
+
+// Base realisation
+func (*BaseManifest) Widgets() []Widget     { return []Widget{} }
+func (*BaseManifest) Actions() []Action     { return []Action{} }
+func (*BaseManifest) Resources() []Resource { return []Resource{} }
+
+func Serve(ctx context.Context, m Manifest, request []byte) {
+	logrus.Debugf("Handling request: %s", string(request))
+	// var f interface{}
+	var err error
+
+	// Starting with widgets
+	widget := &struct {
+		Widget string
+	}{}
+	err = json.Unmarshal(request, widget)
+	if err == nil && widget.Widget != "" {
+		renderedWidget, err := serveWidget(ctx, m, widget.Widget, request)
+		if err != nil {
+			fmt.Print(err)
+			return
+		} else {
+			render(renderedWidget, true)
+			return
+		}
+	}
+
+	// Moving to actions
+	action := &struct {
+		Action string
+	}{}
+	err = json.Unmarshal(request, action)
+	if err == nil && action.Action != "" {
+		err := serveAction(ctx, m, action.Action, request)
+		if err != nil {
+			logrus.Error("Action finished with error: %s", err)
+		}
+		render(nil, true)
+		return
+	}
+
+	// Moving to resource
+	resource := &struct {
+		Resource string
+	}{}
+	err = json.Unmarshal(request, resource)
+	if err == nil && resource.Resource != "" {
+		rawResource, err := serveResource(ctx, m, resource.Resource, request)
+		if err != nil {
+			render(renderManifest(), true)
```

Comment:
```suggestion
			fmt.Print(err)
```
just return also the error for the resource.